presto-native-execution/presto_cpp/main/runtime-metrics/prometheus-reporter/PrometheusReporterUtil.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifdef PRESTO_ENABLE_PROMETHEUS_REPORTER
+#include "presto_cpp/main/runtime-metrics/prometheus-reporter/PrometheusReporter.h"
+#endif
+#include "velox/velox/common/base/StatsReporter.h"
+
+namespace facebook::presto::prometheus {
+constexpr std::string_view kPrometheusReporterNotEnabled{
+    "Prometheus metrics reporter is disabled."};
+
+/// Creates @PrometheusReporter object and returns a unique_ptr for that object.
+/// If Prestissimo is not compiled with PRESTO_ENABLE_PROMETHEUS_REPORTER CMake
+/// flag enabled, this method returns a nullptr.
+std::unique_ptr<velox::BaseStatsReporter> createPrometheusReporter() {
+#ifdef PRESTO_ENABLE_PROMETHEUS_REPORTER
+  auto nodeConfig = NodeConfig::instance();
+  const std::string cluster = nodeConfig->nodeEnvironment();
+  const char* hostName = std::getenv("HOSTNAME");
+  const std::string worker = !hostName ? "" : hostName;
+  std::map<std::string, std::string> labels{
+      {"cluster", cluster}, {"worker", worker}};
+  return std::make_unique<prometheus::PrometheusReporter>(labels);
+#endif
+  return nullptr;
+}
+
+/// Serializes the runtime metrics using the Prometheus data model.
+/// If Prestissimo is not compiled with PRESTO_ENABLE_PROMETHEUS_REPORTER CMake
+/// flag enabled, this method returns an error string.
+const std::string getSerializedMetrics() {
```

Comment:
As mentioned above, I think this function should belong to RuntimeMetricReporter.cpp
The response should be independent of the reporter name.
We don't need the ifdef check here.
```
constexpr std::string_view kMetricsReportersNotRegistered{
    "No metrics reporters are registered."};
auto reporter = folly::Singleton<facebook::velox::BaseStatsReporter>::try_get();
if (auto prometheusReporter =
      std::dynamic_pointer_cast<prometheus::PrometheusReporter>(reporter) ) {
  return prometheusReporter->getSerializedMetrics();
 } else {
  return kMetricsReportersNotRegistered.
 }
```